python/ray/util/client/dataclient.py

```diff
@@ -47,17 +52,20 @@ def __init__(self, channel: "grpc._channel.Channel", client_id: str,
         self.data_thread.start()
 
     def _next_id(self) -> int:
-        with self.lock:
-            self._req_id += 1
-            if self._req_id > INT32_MAX:
-                self._req_id = 1
-            # Responses that aren't tracked (like opportunistic releases)
-            # have req_id=0, so make sure we never mint such an id.
-            assert self._req_id != 0
-            return self._req_id
+        self._req_id += 1
+        if self._req_id > INT32_MAX:
+            self._req_id = 1
```

Comment:
Should we mention that this method requires the Lock to be held?

Reply:
+1.

Reply:
Yes, updated comment.